pkg/inspector/image-inspector.go

```diff
@@ -467,3 +414,153 @@ func createOutputDir(dirName string, tempName string) (string, error) {
+		return nil, "", err
+	}
+	sourceCtx := &types.SystemContext{
+		DockerAuthConfig: nil,
+		DockerCertPath:   certPath,
+	}
+
+	if i.opts.DstPath, err = createOutputDir(i.opts.DstPath, "image-inspector-"); err != nil {
+		return nil, "", err
+	}
+	destRef, err := alltransports.ParseImageName(fmt.Sprintf("dir://%s", i.opts.DstPath))
+	if err != nil {
+		return nil, "", fmt.Errorf("Invalid destination name %s: %v", i.opts.DstPath, err)
+	}
+
+	imagePullAuths, err := i.getAuthConfigs()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	log.Println("Copying image")
+	for name, auth := range imagePullAuths.Configs {
+		sourceCtx.DockerAuthConfig = &types.DockerAuthConfig{
+			Username: auth.Username,
+			Password: auth.Password,
+		}
+		authErr = copy.Image(policyContext, destRef, srcRef, &copy.Options{
+			RemoveSignatures: false,
+			SignBy:           "",
+			ReportWriter:     os.Stdout,
+			SourceCtx:        sourceCtx,
+			DestinationCtx:   nil,
+			ProgressInterval: PULL_LOG_INTERVAL_SEC,
+		})
+		if authErr == nil {
+			break
+		}
+		log.Printf("Authentication with %s failed: %v", name, authErr)
+	}
+
+	if authErr != nil {
+		return nil, "", fmt.Errorf("Unable to pull docker image: %v\n", authErr)
+	}
+
+	img, err := srcRef.NewImage(sourceCtx)
+	if err != nil {
+		return nil, "", err
+	}
+
+	log.Println("Inspecting image")
+	rawManifest, _, err := img.Manifest()
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while reading image manifest: %v\n", err)
+	}
+	imageDigest, err := manifest.Digest(rawManifest)
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while parsing image manifest: %v\n", err)
+	}
+
+	inspectInfo, err := img.Inspect()
+	if err != nil {
+		return inspectInfo, "", fmt.Errorf("Error while inspecting copied image Manifest: %v\n", err)
+	}
+
+	err = i.extractDownloadedImage(inspectInfo)
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while extracting downloaded image: %v\n", err)
+	}
+
+	return inspectInfo, imageDigest, nil
+}
+
+// extractDownloadedImage will untar all the layer tar files specified in 'info'
+// assuming those files exist in i.opts.DstPath .
+func (i *defaultImageInspector) extractDownloadedImage(info *types.ImageInspectInfo) error {
+	for _, layer := range info.Layers {
+		filename := i.opts.DstPath + "/" + strings.SplitN(layer, ":", 2)[1] + ".tar"
+		log.Printf("Untar %s\n", filename)
+		if err := util.UntarGzFile(filename, i.opts.DstPath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// dockerCertPath will try to extract the registry name from the image and return
+// "/etc/docker/certs.d/<RERGISTRY_NAME>" if this path exists or nil otherwise.
+func (i *defaultImageInspector) certPath(fullImageName string) (string, error) {
+	if len(i.opts.RegistryCertPath) > 0 {
+		return i.opts.RegistryCertPath, nil
+	}
+
+	// try to find certificates from docker
+	source_name := strings.SplitN(fullImageName, "://", 2)
```

Comment:
This should not be necessary, containers/image as of https://github.com/containers/image/pull/277 loads the certificates from that directory by default, though _only_ for `docker://` images. (OTOH I don’t think it would be expected for non-`docker:` images to parse `/etc/docker/`* anyway.)

(Also, with the generic transport`:`name syntax acceptable to `alltransports.ParseImageName`, this is not a generally correct way to get the host name; there may not be any host name at all. This should probably use `ImageReference.DockerReference(), if it exists—but really, the defaults should just work nowadays.)